spotbugs/src/main/java/edu/umd/cs/findbugs/config/CommandLine.java

```diff
@@ -342,9 +354,35 @@ private int parse(String argv[], boolean dryRun) throws IOException, HelpRequest
             }
         }
 
+        if (!dryRun && !parsingStdin && parseStdinAsOptions) {
+            String[] stdinArgv = readStdinAsArgv();
+            if (0 != stdinArgv.length) {
+                parse(stdinArgv, true, dryRun);
+            }
+        }
+
         return arg;
     }
 
+    private String[] readStdinAsArgv() throws IOException {
```

Comment:
This implementation loads all data on memory, then its memory-footprint should be large.
Can we reduce this footprint?

Reply:
Redesigning how the ant task parses command-line arguments is outside of the scope of this PR.

I haven't had any issues with memory usage.  I have > 2000 application java classes and 40 external jars.

Reply:
If someone wanted to go about implementing this, I suppose they would need to implement an "ArgumentProvider.getNextArgument()" interface, then have each class which passes arguments to the spotbugs parser implement it.   Then the stdin line reader could read in arguments as needed.

Reply:
Wmm, for me, it seems that this PR just moved the huge memory footprint from the Ant task to the SpotBugs.

Does this change give increase the load for java heap in the SpotBugs side?
If it's limited, then we can put this problem aside for a while, and merge this PR to release this change as a part of 4.1.0.

Reply:
I have no way to compare a run of "it-works-with-this-patch" against "it-crashes-without-this-patch" in order to determine what the possible java heap impact would be.   And honestly, it really doesn't matter to me since a spotbugs run with an potential increased java heap load is far better than a spotbugs run that crashes.

The best fix for ant would be for someone to rewrite the spotbugs engine so that the ant task doesn't have to communicate by passing arguments through the commandline.   That won't help people who run spotbugs from the commandline, though.